### PR TITLE
azure-cli: add `pip` resource

### PR DESCRIPTION
**Pull request by p-linnane**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
`azure-cli` needs `pip` with Python 3.12+ in order for `az extension add` to work: https://github.com/Azure/azure-cli/pull/29922

### CODE REVIEW

File: Formula/a/azure-cli.rb [hunk 1]
```diff
@@ -594,6 +595,11 @@ class AzureCli < Formula
     sha256 "aac08f26a31dc4dffd92821527d1682d99d52f9ef6851968114a8728f3c274d3"
   end
 
+  resource "pip" do
+    url "https://files.pythonhosted.org/packages/4d/87/fb90046e096a03aeab235e139436b3fe804cdd447ed2093b0d70eba3f7f8/pip-24.2.tar.gz"
+    sha256 "5b5e490b5e9cb275c879595064adce9ebd31b854e3e803740b72f9ccf34a45b8"
+  end
```
Comment (carlocab) [inline Formula/a/azure-cli.rb:601]: We don't need a `pip` resource, just do
```diff
diff --git a/Formula/a/azure-cli.rb b/Formula/a/azure-cli.rb
index c09f30b050a..29f819663e7 100644
--- a/Formula/a/azure-cli.rb
+++ b/Formula/a/azure-cli.rb
@@ -719,7 +719,7 @@ class AzureCli < Formula
     ENV["OPENSSL_DIR"] = Formula["openssl@3"].opt_prefix
     ENV["OPENSSL_NO_VENDOR"] = "1"
 
-    venv = virtualenv_create(libexec, "python3.12", system_site_packages: false)
+    venv = virtualenv_create(libexec, "python3.12", system_site_packages: false, without_pip: false)
     venv.pip_install resources
 
     # Get the CLI components we'll install
```

Comment (p-linnane) [reply Formula/a/azure-cli.rb:601]: That's how we had it for Python 3.11, but that's no longer supported:
```
ArgumentError: virtualenv_create's without_pip is deprecated starting with Python 3.12
```

Comment (carlocab) [reply Formula/a/azure-cli.rb:601]: Hm, weird.

**Review by carlocab — APPROVED** [review]

**Comment by bebound** [comment]
@p-linnane Do you know why `setuptools` is not in the venv? It's a dependency of `azure-cli`: https://github.com/Azure/azure-cli/pull/27196/

`/home/linuxbrew/.linuxbrew/Cellar/azure-cli/2.65.0_1/libexec/bin/python -m pip list
| grep setuptools` returns empty.

**Comment by p-linnane** [comment]
@erik-bershel It seems like we need to downgrade back to Python 3.11 despite 3.12 being listed as supported for packaging so that we can unblock you.

@bebound We aren't able to determine the correct resources with the current project set up:
```
╰─ brew update-python-resources -v azure-cli
==> Retrieving PyPI dependencies for "https://github.com/Azure/azure-cli/archive/refs/tags/azure-cli-2.65.0.tar.gz"...
ERROR: https://github.com/Azure/azure-cli/archive/refs/tags/azure-cli-2.65.0.tar.gz does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.
```

So the best thing we can do is run `pipgrip` under Python 3.12 and then manually update the resources.

**Comment by erik-bershel** [comment]
> 3.12 being listed as supported for packaging

Yeah, for now it seams to be very brave statement

**Comment by bebound** [comment]
> @bebound We aren't able to determine the correct resources with the current project set up:
> 
> ```
> ╰─ brew update-python-resources -v azure-cli
> ==> Retrieving PyPI dependencies for "https://github.com/Azure/azure-cli/archive/refs/tags/azure-cli-2.65.0.tar.gz"...
> ERROR: https://github.com/Azure/azure-cli/archive/refs/tags/azure-cli-2.65.0.tar.gz does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.
> ```
> 
> So the best thing we can do is run `pipgrip` under Python 3.12 and then manually update the resources.

I don't know the logic of `brew update-python-resources`. `azure-cli-2.65.0.tar.gz` consists of 3 python projects, and there is no'setup.py' and 'pyproject.toml' in root folder.

---------------

Well, brew calls `pip` with `--no-deps`, I have to manually add `setuptools` and `pip` into resource.

https://github.com/Homebrew/brew/blame/cd651094878e88cb710bc24e253bff8b3595258e/Library/Homebrew/formula.rb#L1855

**Comment by bebound** [comment]
> > 3.12 being listed as supported for packaging
> 
> Yeah, for now it seams to be very brave statement

We've already used 3.12 in Azure Linux 3. Including `pip` and `setuptools` in package is a bit tricky in Homebrew, so we don't bump to 3.12 in 2.65.0. 🥲

**Comment by branchvincent** [comment]
> I don't know the logic of `brew update-python-resources`. `azure-cli-2.65.0.tar.gz` consists of 3 python projects, and there is no 'setup.py' and 'pyproject.toml' in root folder.

I've opened #195315 to ensure `azure-cli` works with our `brew update-python-resources` by changing the formula url to the PyPI sdist `azure-cli` (which depends on the other 2 projects `azure-cli-core` and `azure-cli-telemetry` also on PyPI, meaning they are automatically included as a `resource`). With that change, you no longer need to generate the resources yourself or have a custom script to create version bump PR's, you can use `brew bump` instead. I think this is a far more maintainable solution

**Pull request merged**